src/Dto/StimulusControllersDto.php

```diff
@@ -16,42 +16,20 @@ final class StimulusControllersDto extends AbstractStimulusDto
     private $controllers = [];
     private $values = [];
 
-    /**
-     * @param string $controllerName   the Stimulus controller name
-     * @param array  $controllerValues array of data if a string is passed to the 1st argument
-     *
-     * @throws \Twig\Error\RuntimeError
-     */
-    public function addController($controllerName, array $controllerValues = []): void
+    public function addController(string $controllerName, array $controllerValues = []): void
     {
-        if (\is_string($controllerName)) {
-            $data = [$controllerName => $controllerValues];
-        } else {
-            if ($controllerValues) {
-                throw new \InvalidArgumentException('You cannot pass an array to the first and second argument of stimulus_controller(): check the documentation.');
-            }
-
-            $data = $controllerName;
+        $controllerName = $this->getFormattedControllerName($controllerName);
+        $this->controllers[] = $controllerName;
 
-            if (!$data) {
-                return;
+        foreach ($controllerValues as $key => $value) {
+            if (null === $value) {
+                continue;
             }
-        }
-
-        foreach ($data as $controllerName => $controllerValues) {
-            $controllerName = $this->getFormattedControllerName($controllerName);
-            $this->controllers[] = $controllerName;
 
-            foreach ($controllerValues as $key => $value) {
-                if (null === $value) {
-                    continue;
-                }
+            $key = $this->escapeAsHtmlAttr($this->normalizeKeyName($key));
+            $value = $this->getFormattedValue($value);
 
-                $key = $this->escapeAsHtmlAttr($this->normalizeKeyName($key));
-                $value = $this->getFormattedValue($value);
-
-                $this->values['data-'.$controllerName.'-'.$key.'-value'] = $value;
-            }
+            $this->values['data-'.$controllerName.'-'.$key.'-value'] = $value;
```

Comment:
All of this is covered by tests for controllers, actions and targets (I double-checked).